x/leverage/keeper/msg_server.go

```diff
@@ -187,21 +187,26 @@ func (s msgServer) RepayAsset(
 		return nil, err
 	}
 
-	if err := s.keeper.RepayAsset(ctx, borrowerAddr, msg.Amount); err != nil {
+	repaid, err := s.keeper.RepayAsset(ctx, borrowerAddr, msg.Amount)
+	if err != nil {
 		return nil, err
 	}
 
+	actualRepayment := sdk.NewCoin(msg.Amount.Denom, repaid)
```

Comment:
I think you're trying to differentiate between what was successfully repaid and what the msg wanted to be repaid `msg.Amount` 

```suggestion
	repaidCoin := sdk.NewCoin(msg.Amount.Denom, repaid)
```

Reply:
Changed to `repaidCoin` and/or `rewardCoin` everywhere.